Document metrics emitted by eventing (#15151)
kyma-project/kyma: docs/04-operation-guides/operations/evnt-02-eventing-metrics.md

---
title: Kyma Eventing Metrics
---

> Kyma Eventing emits several custom metrics which are then visualized using [Grafana Dashboards](./evnt-01-eventing-dashboards.md) to monitor statistics and other information for the Eventing backbone in real time.

Reviewer: After replacing "Eventing backbone" with "Kyma Eventing", we can simply drop it actually.

```suggestion
Kyma Eventing provides several Grafana Dashboards, using various [Metrics](./evnt-02-eventing-metrics.md), so you can monitor statistics and other information in real time.
```

### Metrics Emitted by Eventing Publisher Proxy:

| Metric    |  Description |
|-------------|:--------------|
| **event_publish_to_messaging_server_errors_total** | Total number of errors while sending events to the messaging server. |
| **event_publish_to_messaging_server_latency** | Duration of sending events to the messaging server. |
| **event_type_published** | Total number of event publishing requests to the messaging server for a given event type. |
| **event_requests** | Total number of event publishing requests to the messaging server.  |

### Metrics Emitted by Eventing Controller:

| Metric    |  Description |
|-------------|:--------------|
| **event_type_subscribed** | All the eventTypes subscribed using the Subscription CRD. |
| **delivery_per_subscription** | Number of dispatched events per subscription with information regarding the status code and its sink. |

### Metrics Emitted by NATS Exporter:

> The [Prometheus NATS Exporter](https://github.com/nats-io/prometheus-nats-exporter) also emits metrics which can be used for monitoring purposes. More information on NATS Monitoring can be found [here](https://docs.nats.io/running-a-nats-service/configuration/monitoring#jetstream-information).  

Reviewer: ```suggestion
The [Prometheus NATS Exporter](https://github.com/nats-io/prometheus-nats-exporter) also emits metrics that you can monitor. Learn more about [NATS Monitoring](https://docs.nats.io/running-a-nats-service/configuration/monitoring#jetstream-information).  
```
I like active voice and meaningful link titles ;)

---
title: Kyma Eventing Metrics
---

Kyma Eventing emits several custom metrics which are then visualized using [Grafana Dashboards](./evnt-01-eventing-dashboards.md) to monitor statistics and other information for the Eventing backbone in real time.

### Metrics Emitted by Eventing Publisher Proxy:

| Metric    |  Description |
|-------------|:--------------|
| **event_publish_to_messaging_server_errors_total** | Total number of errors while sending events to the messaging server. |
| **event_publish_to_messaging_server_latency** | Duration of sending events to the messaging server. |
| **event_type_published** | Total number of events published for a given event type. |

> | **event_requests** | Total number of event requests for a destination service.  |

Reviewer 2: I am unable to understand this metric. Add more information or rephrase the sentence?

Author: I just used the description from the metric's Help. But I will add more information

Author: https://github.com/kyma-project/kyma/blob/9ee7b1f9bcdf6985a93ed10ed945b7a11ff39dd7/components/event-publisher-proxy/pkg/metrics/collector.go#L26

### Metrics Emitted by Eventing Controller:

| Metric    |  Description |
|-------------|:--------------|
| **event_type_subscribed** | All the eventTypes subscribed using the Subscription CRD. |
| **delivery_per_subscription** | Number of dispatched events per subscription with information regarding the status code and its sink. |

> ### Metrics Emitted by NATS Exporter:

Reviewer 2: Should we document the stuff which is not implemented by us? 
IMO, we can just add a reference to NATS documentation for exposed metrics. Otherwise, we would need to maintain this section every-time NATS changes something.

Author: The reason I added this was because I did not see any of this information (exposed metrics) documented by NATS. The only way for me to search for the metric name was to check the `/metrics` endpoint. The ones I mentioned here are mostly the ones we use in our dashboard

Reviewer 2: yes, this is for user. IMO, we should only communicate the metrics which we are exposing as part of our Eventing services. If some user is interested in NATS metrics, they can see the NATS documentation: https://docs.nats.io/running-a-nats-service/nats_admin/monitoring

title: Kyma Eventing Metrics
---

> Kyma Eventing provides several Grafana Dashboards, using various [Metrics](./evnt-02-eventing-metrics.md), so you can monitor statistics and other information in real time.

Reviewer: ```suggestion
Kyma Eventing provides several Grafana Dashboard with various [metrics](./evnt-02-eventing-metrics.md), so you can monitor statistics and other information in real time.
```
Should be the same as in line 5 of the other doc 😅
